.pre-commit-config.yaml

```diff
@@ -33,6 +45,11 @@ repos:
       args: ['operator-lint']
       pass_filenames: false
 
+- repo: https://github.com/dnephin/pre-commit-golang
```

Comment:
Wait, I thought the whole purpose of this patch is to not use this repo because it's no longer maintained. Isn't it replaced by the golangci-lint hook at L22?

Reply:
I think Alan's comment is correct, we already have a target in the makefile for it (golint i mean) https://github.com/openstack-k8s-operators/cinder-operator/blob/a965c235dec9d6a6674ea82e580c03f07dc7ad43/Makefile#L339
and it uses the openstack-k8s-operators-ci repo

@ASBishop you referred to line 22 but for clarification go-lint and golangci-lint are two different linters 

I see Sandeep requested the addition of go-lint in an earlier review comment here 

on some operators we have both e.g. in ceilometer [1]

```
    - id: make-golangci
      name: make-golangci
      language: system
      entry: make
      args: ['golangci']
AND
    - id: make-golint
      name: make-golint
      language: system
      entry: make
      args: ['golint']
      pass_filenames: false
```

but in keystone we only have golangci-lint [2]
```
    - id: golangci-lint
      name: golangci-lint
      language: golang
      types: [go]
      entry: make
      args: ["golangci-lint"]
      pass_filenames: false
```

In the keystone case we do have a make target for golint [3] but it doesn't seem to be wired up to the hooks in [2]

@Sandeepyadav93 is that correct i.e. should we also enable golint in the hooks for keystone?

[1] https://github.com/openstack-k8s-operators/ceilometer-operator/blob/aa06f9938877798a1744614e43fefc0df8da0eb0/.pre-commit-config.yaml#L27-L45
[2]  https://github.com/openstack-k8s-operators/keystone-operator/blob/b9019b4533f4049d41677329134d6dddb0ab0a63/.pre-commit-config.yaml#L4-L10
[3] https://github.com/openstack-k8s-operators/keystone-operator/blob/b9019b4533f4049d41677329134d6dddb0ab0a63/Makefile#L306

Reply:
While going through https://github.com/golang/lint I found golint is deprecated and frozen now, I think we can start using revive as a replacement for golint.

The Nova team is working on replacing it with revive as well[1].

[1] https://github.com/openstack-k8s-operators/nova-operator/pull/313